src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -0,0 +1,2903 @@
+  return MinimumList;
+}
+
+void MinimumTracer::ReduceVEV(std::vector<double> &vev)
+{
+  // Saveguard if GroupElements is not populated
+  if (GroupElements.size() == 0) return;
+  int MaximumMeasure = -1;
+  char *ptr;
+  std::string BinaryNumber;
+  for (auto GroupElement : GroupElements)
+  {
+    // Clean buffer
+    BinaryNumber.clear();
+    // Feed buffer
+    for (size_t i = 0; i < vev.size(); i++)
+      BinaryNumber.append(std::to_string(vev[i] * GroupElement(i, i) >=
+                                         0)); // Heaviside function
+
+    // Converto to decimal and compare
+    if (std::strtol(BinaryNumber.c_str(), &ptr, 2) > MaximumMeasure)
+    {
+      MaximumMeasure = std::strtol(BinaryNumber.c_str(), &ptr, 2);
+      for (size_t i = 0; i < vev.size(); i++)
+        vev[i] *= GroupElement(i, i); // Transform vev
+    }
+
+    // Reached maximum measure
+    if (MaximumMeasure == pow(2, vev.size()) - 1) return;
+  }
+}
+
+void MinimumTracer::ReduceVEV(Minimum &min)
+{
+  ReduceVEV(min.point);
+}
+
+const std::vector<std::vector<double>>
+MinimumTracer::WarpPath(const std::vector<std::vector<double>> &path,
+                        const std::vector<double> &T1,
+                        const std::vector<double> &F1,
+                        const std::vector<double> &T2,
+                        const std::vector<double> &F2)
+{
+  std::vector<std::vector<double>> r;
+
+  for (std::size_t i = 0; i < path.size(); i++)
+  {
+    std::vector<double> temp = path[i];
+    for (std::size_t d = 0; d < temp.size(); d++)
+    {
+      // translate the path
+      // temp[d] += F2[d] - F1[d]; // TODO: Why commented out? When do we need
+      // to translate the path? warp the path
+      if ((F1[d] - T1[d]) == 0)
+        temp[d] = T2[d] + (temp[d] - T1[d]);
+      else
+        temp[d] = T2[d] + (F2[d] - T2[d]) * (temp[d] - T1[d]) / (F1[d] - T1[d]);
+    }
+    r.push_back(temp);
+  }
+  return r;
+}
+
+MinimumTracer::MinimumTracer()
+{
+}
+
+MinimumTracer::MinimumTracer(
+    const std::shared_ptr<Class_Potential_Origin> &pointer_in,
+    const int &WhichMinimizer_in,
+    const bool &UseMultithreading_in)
+{
+  modelPointer      = pointer_in;
+  WhichMinimizer    = WhichMinimizer_in;
+  UseMultithreading = UseMultithreading_in;
+
+  FindDiscreteSymmetries();
+  FindFlatDirections();
+}
+
+void MinimumTracer::FindFlatDirections()
+{
+  auto nvev      = this->modelPointer->get_nVEV();
+  auto vev_order = this->modelPointer->Get_VevOrder();
+
+  std::vector<double> point = std::vector<double>(nvev, 1);
+
+  // initializsation of vector NonFlatDirections
+  NonFlatDirections = std::vector<int>(nvev, 1);
+
+  double res_1, res_2;
+
+  // 3-dimensional flat directions (SO(3))
+  for (std::size_t i = 0; i < nvev; i++)
+  {
+    for (std::size_t j = i + 1; j < nvev; j++)
+    {
+      for (std::size_t k = j; k < nvev; k++)
+      {
+        point.at(i) = 2;
+        point.at(j) = 100;
+        point.at(k) = 200;
```

Comment:
Where do these numbers come from?

Reply:
These numbers are just an arbitrary point choice to do the flat-directions test.

Reply:
Can we add this as a comment?